core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -676,6 +682,78 @@ public int attachPartition(long timestamp) {
         return StatusCode.OK;
     }
 
+    public StatusCode detachPartition(long timestamp) {
+        // Partitioned table must have a timestamp
+        assert metadata.getTimestampIndex() > -1;
+
+        try {
+            StatusCode statusCode = detachOperationOnPartition(timestamp, false);
+            if (StatusCode.OK != statusCode) {
+                return statusCode;
+            }
+
+            setPathForPartition(path, partitionBy, timestamp, false);
+            if (!ff.exists(path.$())) {
+                LOG.error()
+                        .$("partition folder does not exist [path=")
+                        .$(path)
+                        .$(']')
+                        .$();
+                return StatusCode.PARTITION_FOLDER_DOES_NOT_EXIST;
+            }
+
+            if (inTransaction()) {
```

Comment:
This should go before `detachOperationOnPartition`. Can you add a test which writes but not commits to the partition about to be detached?

Reply:
yes, have done